net.c

```diff
@@ -2970,3 +2978,56 @@ int compare_subnet(char *ip1, char *ip2, unsigned prefix)
   return ok;
 }
 
+
+/*
+ * If there's an 'ifcfg' boot option matching 'device', copy any flags
+ * specified there.
+ *
+ * This is used to supply config.ifcfg.manual with ifcfg flags.
+ */
+void get_and_copy_ifcfg_flags(ifcfg_t *ifcfg, char *device)
+{
+  ifcfg_t *tmp;
+
+  if(!ifcfg || !device) return;
+
+  ifcfg->flags = slist_free(ifcfg->flags);
+
+  if(config.debug >= 2) fprintf(stderr, "ifcfg flags, before(%s):\n%s", device, ifcfg_print(ifcfg));
+
+  // 1st try, direct match
+  for(tmp = config.ifcfg.all; tmp; tmp = tmp->next) {
+    if(tmp->pattern || !tmp->device) continue;
+    if(config.debug >= 2) fprintf(stderr, "direct:\n%s", ifcfg_print(tmp));
```

Comment:
wouldn't improve the situation, imo

There is a long standing plan to have a proper logging function instead of the printf's but that's a bit out of scope here.